ec_encode_data_{sse,avx,avx2} refer to asm routines available only for elf64
--- isa-l-2.29.0/erasure_code/ec_highlevel_func.c.orig	2020-02-27 02:29:49.000000000 +0100
+++ isa-l-2.29.0/erasure_code/ec_highlevel_func.c	2020-05-26 20:43:45.280001729 +0200
@@ -29,7 +29,7 @@
 #include <limits.h>
 #include "erasure_code.h"
 
-#if __x86_64__  || __i386__ || _M_X64 || _M_IX86
+#if __x86_64__  || _M_X64
 void ec_encode_data_sse(int len, int k, int rows, unsigned char *g_tbls, unsigned char **data,
 			unsigned char **coding)
 {
--- isa-l-2.29.0/erasure_code/ec_multibinary.asm.orig	2020-02-27 02:29:49.000000000 +0100
+++ isa-l-2.29.0/erasure_code/ec_multibinary.asm	2020-05-26 21:01:48.970797539 +0200
@@ -36,6 +36,9 @@
  default rel
  [bits 64]
 
+ extern ec_encode_data_sse
+ extern ec_encode_data_avx
+ extern ec_encode_data_avx2
  extern ec_encode_data_update_sse
  extern ec_encode_data_update_avx
  extern ec_encode_data_update_avx2
@@ -62,9 +65,6 @@
 extern gf_vect_dot_prod_sse
 extern gf_vect_dot_prod_avx
 extern gf_vect_dot_prod_avx2
-extern ec_encode_data_sse
-extern ec_encode_data_avx
-extern ec_encode_data_avx2
 
 mbin_interface ec_encode_data
 mbin_interface gf_vect_dot_prod
@@ -73,7 +73,7 @@
 mbin_interface gf_vect_mad
 
 %ifidn __OUTPUT_FORMAT__, elf32
- mbin_dispatch_init5 ec_encode_data, ec_encode_data_base, ec_encode_data_sse, ec_encode_data_avx, ec_encode_data_avx2
+ mbin_dispatch_init2 ec_encode_data, ec_encode_data_base
  mbin_dispatch_init5 gf_vect_dot_prod, gf_vect_dot_prod_base, gf_vect_dot_prod_sse, gf_vect_dot_prod_avx, gf_vect_dot_prod_avx2
  mbin_dispatch_init2 gf_vect_mul, gf_vect_mul_base
  mbin_dispatch_init2 ec_encode_data_update, ec_encode_data_update_base
